### ISSUE TITLE
False positives on interpolated environment variable names

### ISSUE THREAD

**Issue by dgholz** [rhysd/actionlint#312]
I have a workflow like:
```
on: [push]

jobs:
  example:
    runs-on: ubuntu-latest
    steps:
    - name: set output
      id: set-value
      run: |
        echo "value=key_name" >> "$GITHUB_OUTPUT"
    - name: use env var
      env:
        "${{ steps.set-value.outputs.value }}": ${{ github.repository_owner }}
      run: |
        echo "${${{ steps.set-value.outputs.value }}}"
```

(the actual workflow doesn't `echo`, it calls Bundler, which looks for environment variables with special formats to set config options)

This works fine when the workflow runs, but Actionlint reports
```
.github/workflows/test.yml:13:9: environment variable name "${{ steps.set-value.outputs.value }}" is invalid. '&', '=' and spaces should not be contained [env-var]
   |
13 |         "${{ steps.set-value.outputs.value }}": ${{ github.repository_owner }}
   |         ^~~~
```
Deleting the spaces resolves the false positive, but it reveals that Actionlint isn't handling the interpolation in the environment variable name.

**Comment by rhysd** [comment]
I didn't know that `${{ }}` is available for environment variable names. Thanks for reporting this.
